tests/MongoDB.Bson.Tests/Specifications/bson-corpus/BsonCorpusTestRunner.cs

```diff
@@ -0,0 +1,316 @@
+                BsonDocumentSerializer.Instance.Serialize(context, document);
+                return stream.ToArray();
+            }
+        }
+
+        private string EncodeCanonicalExtjson(BsonDocument document)
+        {
+#pragma warning disable 618
+            var writerSettings = new JsonWriterSettings
+            {
+                OutputMode = JsonOutputMode.CanonicalExtendedJson,
+            };
+            if (BsonDefaults.GuidRepresentationMode == GuidRepresentationMode.V2)
+            {
+                writerSettings.GuidRepresentation = GuidRepresentation.Unspecified;
+            }
+#pragma warning restore 618
+            var json = document.ToJson(writerSettings);
+            return json.Replace(" ", "");
+        }
+
+        private string EncodeRelaxedExtjson(BsonDocument document)
+        {
+#pragma warning disable 618
+            var writerSettings = new JsonWriterSettings
+            {
+                OutputMode = JsonOutputMode.RelaxedExtendedJson,
+            };
+            if (BsonDefaults.GuidRepresentationMode == GuidRepresentationMode.V2)
+            {
+                writerSettings.GuidRepresentation = GuidRepresentation.Unspecified;
+            }
+#pragma warning restore 618
+            var json = document.ToJson(writerSettings);
+            return json.Replace(" ", "");
+        }
+
+        private void RunDecodeErrorTest(BsonDocument test)
+        {
+            JsonDrivenHelper.EnsureAllFieldsAreValid(test, "type", "description", "bson");
+
+            var bson = BsonUtils.ParseHexString(test["bson"].AsString);
+
+            var exception = Record.Exception(() =>
+            {
+                using (var stream = new MemoryStream(bson))
+                using (var reader = new BsonBinaryReader(stream))
+                {
+                    while (!reader.IsAtEndOfFile())
+                    {
+                        _ = BsonSerializer.Deserialize<BsonDocument>(reader);
+                    }
+                }
+            });
+
+            exception.Should().NotBeNull();
+        }
+
+        private void RunParseErrorTest(BsonDocument test)
+        {
+            JsonDrivenHelper.EnsureAllFieldsAreValid(test, "type", "description", "string");
+
+            var json = test["string"].AsString;
+
+            var exception = Record.Exception(() => BsonDocument.Parse(json));
+
+            exception.Should().NotBeNull();
+        }
+
+        private void RunValidTest(BsonDocument test)
+        {
+            JsonDrivenHelper.EnsureAllFieldsAreValid(
+                test,
+                "type",
+                "description",
+                "canonical_bson",
+                "canonical_extjson",
+                "relaxed_extjson",
+                "degenerate_bson",
+                "degenerate_extjson",
+                "converted_bson",
+                "converted_extjson",
+                "lossy");
+
+            byte[] cB = null;
```

Comment:
Normally abbreviations like this should not make it through code review, but in this case since these are the names that are used in the spec we can use them.